pygeos/_geos.pyx

```diff
@@ -0,0 +1,16 @@
+cdef class get_geos_handle:
+    '''This class provides a context manager that wraps the GEOS context handle.
+
+    Example
+    -------
+    with get_geos_handle() as geos_handle:
+        SomeGEOSFunc(geos_handle, ...<other params>)
+    '''
+    cdef GEOSContextHandle_t __enter__(self):
+        self.handle = GEOS_init_r()
+        return self.handle
+
+    def __exit__(self, type, value, traceback):
+        GEOS_finish_r(self.handle)
```

Comment:
I am wondering, would we be able to just put this in `_geos.pxd` as well? (to avoid having this additional pyx file)

Reply:
My interpretation of the [docs](https://cython.readthedocs.io/en/latest/src/userguide/sharing_declarations.html#sharing-extension-types) was that these had to be split between files for extension types (`cdef class`), at least when we need Python methods.  Otherwise we get compile errors or errors in usage (compiler lets us do a `cdef inline __exit__` but then it fails on usage with an `AttributeError`)

Reply:
Yeah, reading the docs, it seems you can put code in pxd file, but that only is useful if it can be inlined, which is certainly not the case here anyway (given it's not a full cdef)